slither/core/declarations/structure.py

```diff
@@ -51,3 +51,12 @@ def elems_ordered(self) -> List["StructureVariable"]:
 
     def __str__(self) -> str:
         return self.name
+
+    def __eq__(self, other) -> bool:
+        if not isinstance(other, Structure):
+            return False
+        for idx, elem in enumerate(self.elems_ordered):
+            other_elem = other.elems_ordered[idx]
+            if other_elem.type != elem.type or other_elem.name != elem.name:
```

Comment:
This can cause an infinite loop because the elem can be a user defined type and its __eq__ calls back into structure when comparing its underlying type
https://github.com/crytic/slither/blob/master/tests/function_ids/rec_struct-0.8.sol

```
slither/core/solidity_types/user_defined_type.py:77: in __eq__
    return self.type == other.type
slither/core/declarations/structure.py:60: in __eq__
    if other_elem.type != elem.type or other_elem.name != elem.name:
```

Reply:
Ahh I see that now. Hmmm, I'll have to think up a way to avoid that. Maybe we just skip over any element in the structure which is an instance of the same structure? But that might still lead to an infinite loop if the element is an instance of some other structure, and that structure includes an element which is an instance of the original struct...